raiden/tests/integration/test_echo_node.py

```diff
@@ -0,0 +1,197 @@
+    address_to_app = {app.raiden.address: app for app in raiden_chain}
+    token_address = token_addresses[0]
+    echo_api = RaidenAPI(echo_app.raiden)
+
+    echo_node = EchoNode(echo_api, token_address)
+    echo_node.ready.wait(timeout=30)
+    assert echo_node.ready.is_set()
+
+    expected = list()
+
+    # Create some transfers
+    for num, app in enumerate([app0, app1, app2]):
+        amount = 1 + num
+        transfer_event = RaidenAPI(app.raiden).transfer_async(
+            token_address,
+            amount,
+            echo_app.raiden.address,
+            10 ** (num + 1)
+        )
+        transfer_event.wait(timeout=20)
+        expected.append(amount)
+
+    while echo_node.num_handled_transfers < len(expected):
+        gevent.sleep(.5)
+
+    # Check that all transfers were handled correctly
+    for handled_transfer in echo_node.seen_transfers:
+        app = address_to_app[handled_transfer['initiator']]
+        events = get_channel_events_for_token(app, token_address, 0)
+        received = {}
+
+        for event in events:
+            if event['_event_type'] == 'EventTransferReceivedSuccess':
+                received[repr(event)] = event
+
+        assert len(received) == 1
+        transfer = received.values()[0]
+        assert transfer['initiator'] == echo_app.raiden.address
+        assert transfer['identifier'] == (
+            handled_transfer['identifier'] + transfer['amount']
+        )
+
+    echo_node.stop()
+
+
+# `RaidenAPI.get_channel_events` is not supported in tester
+@pytest.mark.parametrize('blockchain_type', ['geth'])
+@pytest.mark.parametrize('number_of_nodes', [8])
+@pytest.mark.parametrize('number_of_tokens', [1])
+@pytest.mark.parametrize('channels_per_node', [CHAIN])
+@pytest.mark.parametrize('reveal_timeout', [20])
+@pytest.mark.parametrize('settle_timeout', [120])
+def test_echo_node_lottery(
+    token_addresses,
+    raiden_chain,
+):
+    app0, app1, app2, app3, echo_app, app4, app5, app6 = raiden_chain
+    address_to_app = {app.raiden.address: app for app in raiden_chain}
+    token_address = token_addresses[0]
+    echo_api = RaidenAPI(echo_app.raiden)
+
+    echo_node = EchoNode(echo_api, token_address)
+    echo_node.ready.wait(timeout=30)
+    assert echo_node.ready.is_set()
+
+    expected = list()
+
+    # Let 6 participants enter the pool
+    amount = 7
+    for num, app in enumerate([app0, app1, app2, app3, app4, app5]):
+        transfer_event = RaidenAPI(app.raiden).transfer_async(
+            token_address,
+            amount,
+            echo_app.raiden.address,
+            10 ** (num + 1)
+        )
+        transfer_event.wait(timeout=20)
+        expected.append(amount)
+
+    # test duplicated identifier + amount is ignored
+    transfer_event = RaidenAPI(app5.raiden).transfer_async(
+        token_address,
+        amount,
+        echo_app.raiden.address,
+        10 ** 6
+    ).wait(timeout=20)
+    # test pool size querying
+    pool_query_identifier = 77
```

Comment:
`pool_query_identifier` has no special value here? As long as the amount is `7` and the initiator has already sent something to the echo node before he will get back the response identifier `identifier+len(pool)` and response amount `len(pool)`?